src/bin/railroad.rs

```diff
@@ -0,0 +1,47 @@
+extern crate railroad_dsl;
+extern crate structopt;
+
+use std::error::Error;
+use std::fs;
+use std::path::PathBuf;
+use std::io::{self, Read};
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+#[structopt(about = "Process railroad diagrams according to DSL.
+
+If no input files are given, act as a pipe from stdin to stdout.
+Otherwise, process each input file into an output file with
+the file extension replaced by `.svg`.")]
+struct Options {
+    #[structopt(help = "Input files to process")]
+    inputs: Vec<String>,
+}
+
+
+fn main() -> Result<(), Box<Error>> {
+    let args = Options::from_args();
+    if args.inputs.is_empty() {
+        let mut buf = String::new();
+        io::stdin().read_to_string(&mut buf)?;
+        match railroad_dsl::compile(&buf) {
+            Err(e) => eprintln!("syntax error:\n{}", e),
+            Ok((_, _, diagram)) => println!("{}", diagram),
+        }
+    } else {
+        for input in args.inputs {
+            let output = PathBuf::from(&input).with_extension("svg");
+            match fs::read_to_string(&input) {
+                Err(e) => eprintln!("error reading file {}: {}", input, e),
+                Ok(buf) => match railroad_dsl::compile(&buf) {
+                    Err(e) => eprintln!("syntax error:\n{}", e),
```

Comment:
Maybe we should print the filename here as well, so it's clear where the error is coming from